File: VB/ToggleSwitch/ToggleSwitch/RepositoryItemToggleSwitch.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository

Namespace CustomEditor
	<System.ComponentModel.DesignerCategory(""), UserRepositoryItem("Register")> _
	Public Class RepositoryItemToggleSwitch
		Inherits RepositoryItem
		Shared Sub New()

			Register()
		End Sub
		Public Sub New()
			MyBase.New()

		End Sub


		Friend Const EditorName As String = "ToggleSwitch"

		Public Shared Sub Register()
            EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(ToggleSwitch), GetType(RepositoryItemToggleSwitch), GetType(ToggleSwitchViewInfo), New ToggleSwitchPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property


	End Class
End Namespace

File: CS/ToggleSwitch/ToggleSwitch/RepositoryItemToggleSwitch.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;

namespace CustomEditor {
    [System.ComponentModel.DesignerCategory("")]
    [UserRepositoryItem("Register")]
    public class RepositoryItemToggleSwitch : RepositoryItem{
        static RepositoryItemToggleSwitch() {

            Register();
        }
        public RepositoryItemToggleSwitch()
            : base() {
            
        }


        internal const string EditorName = "ToggleSwitch";

        public static void Register() {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(ToggleSwitch),
                typeof(RepositoryItemToggleSwitch), typeof(ToggleSwitchViewInfo),
                new ToggleSwitchPainter(), true, null));
        }
        public override string EditorTypeName {
            get { return EditorName; }
        }


    }
}
